src/detect-http-header.c

```diff
@@ -209,14 +214,8 @@ static uint8_t DetectEngineInspectBufferHttpHeader(DetectEngineCtx *de_ctx,
         return DETECT_ENGINE_INSPECT_SIG_MATCH;
     }
 end:
-    if (flags & STREAM_TOSERVER) {
```

Comment:
this change looks like it possibly changes behavior? No explicit HTTP1 anymore... probably deserves its own commit and explanation

Reply:
> this change looks like it possibly changes behavior?

It should not :
- DetectEngineInspectBufferHttpHeader is only used with ALPROTO_HTTP1
- engine->progress should be HTP_REQUEST_HEADERS or HTP_RESPONSE_HEADERS based on the direction 

> probably deserves its own commit and explanation

ok